Abhi-AnimeX/app/src/main/java/com/abhijith/animex/ui/screens/animelist/AnimeList.kt

```diff
@@ -1,18 +1,36 @@
+package com.abhijith.animex.ui.screens.animelist
+
+import android.net.Uri
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.lifecycle.viewmodel.compose.viewModel
-import com.abhijith.animex.ui.screens.animelist.AnimeListItem
+import androidx.navigation.NavController
+import com.abhijith.animex.ui.screens.Screen
 import com.abhijith.animex.ui.screens.animelist.viewmodel.AnimeListViewModel
+import com.google.gson.Gson
 
 @Composable
-fun AnimeList(viewModel: AnimeListViewModel = viewModel()) {
-    val animeItems = viewModel.items.collectAsState()
+fun AnimeList(navController: NavController, viewModel: AnimeListViewModel = viewModel()) {
+    val animeItems by viewModel.items.collectAsState()
+    val navigationEvent by viewModel.navigationEvent.collectAsState()
+
+    LaunchedEffect(navigationEvent) {
+        navigationEvent?.let { anime ->
+            val animeJson = Uri.encode(Gson().toJson(anime))
+            navController.navigate(
+                Screen.AnimeDetails.route.replace("{selectedAnimeItem}", animeJson))
+            // Reset the navigation event
+            viewModel.onResetNavigation()
+        }
```

Comment:
AnimeList should not have this much dependency

Reply:
Not sure how do i fix this ... can you guide me here?
do you want me to bring it out to AppNavHost or should i move this to viewModel?
@anandwana001